bbreaker/GameObject.py

```diff
@@ -87,34 +113,20 @@ def handle_events(self):
                 self.handle_mouse_events(event)
 
     def handle_mouse_events(self, event):
-        if event.type == pygame.MOUSEBUTTONDOWN:
-            pygame.mouse.set_cursor(*pygame.cursors.broken_x)
-            mouseButtons = pygame.mouse.get_pressed()
-            if mouseButtons[0] == 1:
-                self.start_pos = event.pos
-
-        if event.type == pygame.MOUSEBUTTONUP:
-            pygame.mouse.set_cursor(*pygame.cursors.arrow)
-            if self.start_pos and pygame.mouse.get_pressed()[0] == 0:
-                self.end_pos = event.pos
-                self.strike()
-
-        if event.type == pygame.MOUSEMOTION:
-            if pygame.mouse.get_pressed()[0] == 1:
-                # pressed - draw line shot direction
-                pass
+        self.current_mpos = pygame.mouse.get_pos()
 
-    def strike(self):
-        if self.end_pos[1] < self.start_pos[1]:
-            return
+        if event.type == pygame.MOUSEBUTTONDOWN and Ball.init_position[0]:
+            self.strike()
 
-        alfa = Ball.calculate_strike_angle(self.start_pos, self.end_pos)
-        min_angle = math.radians(10)
-        if alfa < -math.pi + min_angle or alfa > -min_angle:
+        self.update()
+
+    def strike(self):
+        alfa = Ball.calculate_strike_angle(Ball.init_position, self.current_mpos)
+        if not (math.radians(170) >= alfa >= math.radians(10)):
             return
 
-        Ball.init_velocity[0] = round(Ball.init_speed * math.cos(alfa))
-        Ball.init_velocity[1] = round(Ball.init_speed * math.sin(alfa))
+        Ball.init_velocity[0] = Ball.init_speed * math.cos(alfa)
+        Ball.init_velocity[1] = -Ball.init_speed * math.sin(alfa)
```

Comment:
Can we move a ball by ex. 4.8 px? Is the fraction automatically truncated when a position is set? I did not test it, just asking.